### PR TITLE
Extend Cipher types

### PR DESCRIPTION
**Pull request by Taowyoo**
This PR is based on https://github.com/fortanix/rust-mbedtls/pull/327
So please only review the commits start from: [697923d](https://github.com/fortanix/rust-mbedtls/pull/330/commits/697923d33fdb19ea6d87643c8b8202e98d3f9c45)

### CODE REVIEW

**Comment by Taowyoo** [comment]
I revert the `feat: make Hmac clone-able` because the normal clone logic does not work well with Hmac .

File: mbedtls/src/cipher/raw/mod.rs [hunk 1]
```diff
@@ -135,6 +141,31 @@ define!(
         Camellia128Ccm = CIPHER_CAMELLIA_128_CCM,
         Camellia192Ccm = CIPHER_CAMELLIA_192_CCM,
         Camellia256Ccm = CIPHER_CAMELLIA_256_CCM,
+        CipherAria128Ecb = CIPHER_ARIA_128_ECB,
```
[the remark this answers is not in the document]
Comment (Taowyoo) [reply mbedtls/src/cipher/raw/mod.rs:145]: sure, will fix it

File: mbedtls/src/cipher/raw/mod.rs [hunk 2]
```diff
@@ -39,6 +40,7 @@ impl From<cipher_id_t> for CipherId {
             CIPHER_ID_BLOWFISH => CipherId::Blowfish,
             CIPHER_ID_ARC4 => CipherId::Arc4,
             CIPHER_ID_ARIA => CipherId::Aria,
+            CIPHER_ID_CHACHA20 => CipherId::Aria,
```
[the remark this answers is not in the document]
Comment (Taowyoo) [reply mbedtls/src/cipher/raw/mod.rs:44]: nice catch

[the remark this answers is not in the document]
Comment (Taowyoo) [reply mbedtls/src/cipher/raw/mod.rs]: fixed

File: mbedtls/src/cipher/raw/mod.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (Taowyoo) [reply mbedtls/src/cipher/raw/mod.rs]: fixed

File: mbedtls/src/ecp/mod.rs [hunk 3]
```diff
@@ -283,6 +283,12 @@ impl EcPoint {
         }
     }
 
+    pub fn from_binary_no_compress(group: &EcGroup, bin: &[u8]) -> Result<EcPoint> {
```
Comment (s-arash) [inline mbedtls/src/ecp/mod.rs:286]: I don't see it used anywhere. Do we need it?

**Review by s-arash — APPROVED** [review]

File: mbedtls/src/ecp/mod.rs (hunk 3 above)
Comment (s-arash) [reply mbedtls/src/ecp/mod.rs:286]: Oh I see.

**Pull request merged**